barterdude/hooks/__init__.py

```diff
@@ -23,7 +22,7 @@ async def on_connection_fail(self, error: Exception, retries: int):
 
 
 class HttpHook(BaseHook):
-    def __init__(self, barterdude: BarterDude, path: str):
+    def __init__(self, barterdude, path: str):
```

Comment:
because python typing is new and sometimes get cyclic dependency and just python 3.7 "solves" it, but Barterdude supports 3.6 and break compatibility just because of it is not fair: https://stackoverflow.com/questions/33837918/type-hints-solve-circular-dependency